expression/builtin_json.go

```diff
@@ -743,8 +745,124 @@ type jsonSearchFunctionClass struct {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_SEARCH")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	// json_doc, one_or_all, search_str[, escape_char[, path] ...])
+	argTps := make([]types.EvalType, 0, len(args))
+	argTps = append(argTps, types.ETJson)
+	for range args[1:] {
+		argTps = append(argTps, types.ETString)
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETJson, argTps...)
+	sig := &builtinJSONSearchSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_JsonSearchSig)
+	return sig, nil
+}
+
+func (b *builtinJSONSearchSig) evalJSON(row chunk.Row) (res json.BinaryJSON, isNull bool, err error) {
+	// json_doc
+	var obj json.BinaryJSON
+	obj, isNull, err = b.args[0].EvalJSON(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+
+	// one_or_all
+	var containType string
+	containType, isNull, err = b.args[1].EvalString(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+	if containType != json.ContainsPathAll && containType != json.ContainsPathOne {
+		return res, true, json.ErrInvalidJSONContainsPathType
+	}
+
+	// search_str & escape_char
+	var searchStr string
+	searchStr, isNull, err = b.args[2].EvalString(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+	escape := byte('\\')
+	if len(b.args) >= 4 {
+		var escapeStr string
+		escapeStr, isNull, err = b.args[3].EvalString(b.ctx, row)
+		if err != nil {
+			return res, isNull, errors.Trace(err)
+		}
+		if isNull || len(escapeStr) == 0 {
+			escape = byte('\\')
+		} else if len(escapeStr) == 1 {
+			escape = byte(escapeStr[0])
+		} else {
+			err = errors.New("Incorrect arguments to ESCAPE")
+			return res, true, errors.Trace(err)
+		}
+	}
+	patChars, patTypes := stringutil.CompilePattern(searchStr, escape)
+
+	// result
+	result := make([]interface{}, 0)
+	resultSet := make(map[string]bool)
+
+	// walk json_doc
+	walkFn := func(fullpath json.PathExpression, bj json.BinaryJSON) (stop bool, err error) {
+		pathStr := fullpath.String()
+		if _, ok := resultSet[pathStr]; ok {
+			return false, nil
+		}
+		if bj.TypeCode == json.TypeCodeString && stringutil.DoMatch(string(bj.GetString()), patChars, patTypes) {
+			resultSet[pathStr] = true
+			result = append(result, pathStr)
+			if containType == json.ContainsPathOne {
+				return true, nil
+			}
+		} else {
+			resultSet[pathStr] = false
+		}
+		return false, nil
+	}
+	if len(b.args) >= 5 { // path...
+		pathExprs := make([]json.PathExpression, 0, len(b.args)-4)
+		for i := 4; i < len(b.args); i++ {
+			var s string
+			s, isNull, err = b.args[i].EvalString(b.ctx, row)
+			if isNull || err != nil {
+				return res, isNull, errors.Trace(err)
+			}
+			var pathExpr json.PathExpression
+			pathExpr, err = json.ParseJSONPathExpr(s)
+			if err != nil {
+				return res, true, errors.Trace(err)
+			}
+			pathExprs = append(pathExprs, pathExpr)
+		}
+		obj.Walk(walkFn, pathExprs...)
```

Comment:
I am excited to see `json_search`, so I checked out your branch :-)  The CI is broken because an error returned by this line, and line 854 are not captured:

`err = obj.Walk(walkFn, pathExprs...)` should fix it.